### ISSUE TITLE
Voiceover

### ISSUE THREAD

**Issue by dan-gittik** [dan-gittik/canim#11]
Add support for voiceover. It should be enabled with the `voiceover` configuration (in the `code_animation` decorator and the `CodeConfig` class; default is `False`), and be available via a context manager similar to that of Manim's `voiceover`:

```python
@code_animation(voiceover=True)
def with_voiceover(scene: CodeScene):
    block = scene.create_code_block(-6, 3)
    with scene.voiceover('Hello, world!'):
        block.append_lines('Hello, world!')
```

The `voiceover` context generator should yield a tracker, but in most cases it shouldn't be necessary; instead, we're going to use bookmarks to synchronize the animation. Bookmarks are inserted as `{tag}`s like this, and then awaited with `scene @ 'tag'`; for example:

```python
@code_animation(voiceover=True)
def with_voiceover(scene: CodeScene):
    block = scene.create_code_block(-6, 3)
    with scene.voiceover('''
        Line 1{1}
        Line 2{2}
        Line 3
    ''')
        block.append_lines('Line 1')
        scene @ 1
        block.append_lines('Line 2')
        scene @ 2
        block.append_lines('Line 3')
```

**Comment by dan-gittik** [comment]
This is really easy, thanks to Manim's builtin voiceover capabilities. All we need to do is install it (including PortAudio and Sox as system dependencies, and `manim-voiceover`, `manim-voiceover[recorder]` and `manim-voiceover[transcribe]` as requirements), and use a `VoiceoverScene` instead of a `Scene`. If the voiceover configuration is active, we should `set_speech_service` to `RecorderService`, and the `voiceover` wrapper should simply delegate to Manim's after substituting anything of the regex `\{(.*?)\}` with `<bookmark mark="\1">`. Other than that, the matrix multiplication operation should be overloaded to call `wait_until_bookmark`.
